doc/src/devdocs/gc.md

```diff
@@ -2,77 +2,60 @@
 
 ## Introduction
 
-Julia has a serial, stop-the-world, generational, non-moving mark-sweep garbage collector.
-Native objects are precisely scanned and foreign ones are conservatively marked.
+Julia has a non-moving, partially concurrent, parallel, generational and mostly precise mark-sweep collector (an interface
+for conservative stack scanning is provided as an option for users who wish to call Julia from C).
 
-## Memory layout of objects and GC bits
+## Allocation
 
-An opaque tag is stored in the front of GC managed objects, and its lowest two bits are
-used for garbage collection.  The lowest bit is set for marked objects and the second
-lowest bit stores age information (e.g. it's only set for old objects).
+Julia uses two types of allocators, the size of the allocation request determining which one is used. Objects up to 2k
+bytes are allocated on a per-thread free-list pool allocator, while objects larger than 2k bytes are allocated through libc
+malloc.
 
-Objects are aligned by a multiple of 4 bytes to ensure this pointer tagging is legal.
+Julia’s pool allocator partitions objects on different size classes, so that a memory page managed by the pool allocator
```

Comment:
Sorry I didn't get to reviewing this before the holiday! Looks great from what I've read so far.

One question: Do you think we could use this opportunity to change the terminology from "GC Page" or "GC memory page" to "GC Block"? It seems unnecessarily confusing to use Page for both OS pages and GC pages, and we've talked about wanting to change that for a while. :) Is this rewrite a good opportunity to do that?